### PR TITLE
docs: clarified the behavior when waited timeout.

### PR DESCRIPTION
**Pull request by spacewander**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -49,7 +49,11 @@ faster.
 When stdin is nil or an empty string, the stdin device will get closed immediately.
 
 The `timeout` parameter speifies the timeout threshold for stderr/stdout reading timeout,
-the stdin writing timeout, and the process waiting timeout, respectively.
+the stdin writing timeout, and the process waiting timeout, respectively. If the timeout
+expires, the sub-process will be killed by first a SIGTERM, and then a SIGKILL.
```
Comment (agentzh) [inline README.md]: "the timeout expires" does not make sense. Should be "the timeout threshold is exceeded"?

Also, we need to make it clear about

1. the `SIGKILL` won't be issued when `SIGTERM` already makes the process quit.
1. the waiting time between `SIGTERM` and `SIGKILL`.
1 whether we always wait the full delay between `SIGTERM` and `SIGKILL` even if the process quits almost immediately upon receiving the `SIGTERM` signal.

File: README.md [hunk 2]
```diff
@@ -49,7 +49,11 @@ faster.
...
+
+Note that we do nothing with the child processes of the sub-process (if there are),
```
Comment (agentzh) [inline README.md]: "do nothing" is ambiguous. We should say "do not automatically kill the sub-process".

**Review by agentzh — CHANGES_REQUESTED** [review]

**Comment by agentzh** [comment]
"when waited timeout" is not proper English in the title, BTW. Maybe should be "when waiting is timed out"? Does it only apply to `:wait()` at all? It's not clear to me.

**Comment by spacewander** [comment]
@agentzh 
Updated, and yes, the clean up only happens when waiting timed out.

File: README.md [hunk 3]
```diff
@@ -49,7 +49,12 @@ faster.
 When stdin is nil or an empty string, the stdin device will get closed immediately.
 
 The `timeout` parameter speifies the timeout threshold for stderr/stdout reading timeout,
-the stdin writing timeout, and the process waiting timeout, respectively.
+the stdin writing timeout, and the process waiting timeout, respectively. If the timeout
+threshold is exceeded, the sub-process will be killed by first a SIGTERM. Then we will
```
Comment (agentzh) [inline README.md:53]: "by first a SIGTERM" is not proper English. I think you can write this:

> will be killed like this:
> 
> 1. first send a SIGTERM signal, and then
> 2. after 1 millisecond, sending a SIGKILL signal.

BTW, I do believe `1ms` is too soon. Many processes may take much longer than 1ms to clean up. How about 100ms instead?

File: README.md [hunk 4]
```diff
@@ -49,7 +49,12 @@ faster.
...
+wait for 1ms and send a SIGKILL to kill the sub-process.
+
+Note that we do not automatically kill the child processes of the sub-process (if there are),
```
Comment (agentzh) [inline README.md:56]: "if there are" => "if any".

**Review by agentzh — CHANGES_REQUESTED** [review]

**Comment by thibaultcha** [comment]
Merged with the proper edits. Thanks!

**Pull request closed without merging**